From 0a9787897f3d36055130b0532967d31c31e8408f Mon Sep 17 00:00:00 2001
From: K900 <me@0upti.me>
Date: Sun, 9 Mar 2025 09:37:33 +0300
Subject: [PATCH] Fix build with Pipewire 1.4

Pipewire 1.4 changed some functions from taking void*
to taking the actual type they expect, which makes
some code that built with 1.2 no longer build with 1.4.

For older versions, this is effectively a no-op.

Change-Id: Ia51e1695c418a2c8064818756348710bf272453a
Bug: webrtc:401732633
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/380500
Reviewed-by: Jan Grulich <grulja@gmail.com>
Commit-Queue: Jan Grulich <grulja@gmail.com>
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Reviewed-by: Ilya Nikolaevskiy <ilnik@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#44076}
---
 modules/video_capture/linux/pipewire_session.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/video_capture/linux/pipewire_session.cc b/modules/video_capture/linux/pipewire_session.cc
index b55fac188e..a5c568240b 100644
--- a/modules/video_capture/linux/pipewire_session.cc
+++ b/modules/video_capture/linux/pipewire_session.cc
@@ -87,7 +87,7 @@ PipeWireNode::PipeWireNode(PipeWireSession* session,
       .param = OnNodeParam,
   };
 
-  pw_node_add_listener(proxy_, &node_listener_, &node_events, this);
+  pw_node_add_listener(reinterpret_cast<pw_node*>(proxy_), &node_listener_, &node_events, this);
 }
 
 // static
@@ -119,7 +119,7 @@ void PipeWireNode::OnNodeInfo(void* data, const pw_node_info* info) {
       uint32_t id = info->params[i].id;
       if (id == SPA_PARAM_EnumFormat &&
           info->params[i].flags & SPA_PARAM_INFO_READ) {
-        pw_node_enum_params(that->proxy_, 0, id, 0, UINT32_MAX, nullptr);
+        pw_node_enum_params(reinterpret_cast<pw_node*>(that->proxy_), 0, id, 0, UINT32_MAX, nullptr);
         break;
       }
     }
